src/test_utils.rs

```diff
@@ -29,49 +31,110 @@ use crate::dc_receive_imf::dc_receive_imf;
 use crate::dc_tools::EmailAddress;
 use crate::events::{Event, EventType};
 use crate::job::Action;
-use crate::key::{self, DcKey};
+use crate::key::{self, DcKey, KeyPair, KeyPairUse};
 use crate::message::{update_msg_state, Message, MessageState, MsgId};
 use crate::mimeparser::MimeMessage;
 use crate::param::{Param, Params};
 
 #[allow(non_upper_case_globals)]
 pub const AVATAR_900x900_BYTES: &[u8] = include_bytes!("../test-data/image/avatar900x900.png");
 
-type EventSink =
-    dyn Fn(Event) -> Pin<Box<dyn Future<Output = ()> + Send + 'static>> + Send + Sync + 'static;
-
 /// Map of [`Context::id`] to names for [`TestContext`]s.
 static CONTEXT_NAMES: Lazy<std::sync::RwLock<BTreeMap<u32, String>>> =
     Lazy::new(|| std::sync::RwLock::new(BTreeMap::new()));
 
+#[derive(Debug, Clone, Default)]
+pub struct TestContextBuilder {
+    name: Option<String>,
```

Comment:
Clippy (actually the compiler) complains about unused field, same for `addr`.

Reply:
seems fair, but i think this needs to be clearer: https://github.com/deltachat/deltachat-core-rust/pull/2926